mod/file/actions/file/upload.php

```diff
@@ -42,25 +64,18 @@
 	}
 }
 
-if (!elgg_is_empty($title)) {
-	$file->title = $title;
+foreach ($values as $name => $value) {
+	$file->{$name} = $value;
 }
 
-$file->description = $desc;
-$file->access_id = $access_id;
-$file->container_guid = $container_guid;
-$file->tags = elgg_string_to_array($tags);
-
-$file->save();
+if (!$file->save()) {
+	return elgg_error_response(elgg_echo('file:uploadfailed'));
+}
 
 if ($uploaded_file && $uploaded_file->isValid()) {
 	if (!$file->acceptUploadedFile($uploaded_file)) {
 		return elgg_error_response(elgg_echo('file:uploadfailed'));
 	}
-	
-	if (!$file->save()) {
```

Comment:
isn't this needed to have a time updated after the file upload? maybe for cache flushes